### PR TITLE
fix cvtColor-error

### PR DESCRIPTION
**Pull request by UncleLLD**
fix the iusse: https://github.com/opencv/opencv/issues/21223

* Added a conditional sentence to judge whether ret is true
* Modify the problem of using the `next` keyword as a variable
* some python code style: add space

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: samples/python/tutorial_code/video/optical_flow/optical_flow.py [hunk 1]
```diff
@@ -33,30 +33,35 @@
...
-        mask = cv.line(mask, (int(a),int(b)),(int(c),int(d)), color[i].tolist(), 2)
-        frame = cv.circle(frame,(int(a),int(b)),5,color[i].tolist(),-1)
-    img = cv.add(frame,mask)
-
-    cv.imshow('frame',img)
-    k = cv.waitKey(30) & 0xff
-    if k == 27:
+    ret, frame = cap.read()
+    if ret:
+        frame_gray = cv.cvtColor(frame, cv.COLOR_BGR2GRAY)
```
Comment (alalek) [inline samples/python/tutorial_code/video/optical_flow/optical_flow.py:38]: Please reuse code pattern from already existed in other samples (e.g. [here](https://github.com/opencv/opencv/blob/9bcb006ec7b97a076b289acb6fe7375b4c8d933f/samples/dnn/face_detect.py#L138-L141))

```
ret, frame = cap.read()
if not ret:
    ... message...
    waitKey(0)
    break

... processing code kept here ...
```

File: samples/python/tutorial_code/video/optical_flow/optical_flow_dense.py [hunk 2]
```diff
@@ -2,22 +2,27 @@
...
-    mag, ang = cv.cartToPolar(flow[...,0], flow[...,1])
-    hsv[...,0] = ang*180/np.pi/2
-    hsv[...,2] = cv.normalize(mag,None,0,255,cv.NORM_MINMAX)
-    bgr = cv.cvtColor(hsv,cv.COLOR_HSV2BGR)
-    cv.imshow('frame2',bgr)
-    k = cv.waitKey(30) & 0xff
-    if k == 27:
+    if ret:
+        frame2_gray = cv.cvtColor(frame2, cv.COLOR_BGR2GRAY)
+        flow = cv.calcOpticalFlowFarneback(frame1_gray, frame2_gray, None, 0.5, 3, 15, 3, 5, 1.2, 0)
```
Comment (alalek) [inline samples/python/tutorial_code/video/optical_flow/optical_flow_dense.py:12]: > prvs,next
> frame1_gray, frame2_gray

Please don't rename vars (this introduces unnecessary changes in the patch)

**Review by alalek — COMMENTED** [review]
Thank you for contribution!

File: samples/python/tutorial_code/video/optical_flow/optical_flow_dense.py (hunk 2 above)
Comment (UncleLLD) [reply samples/python/tutorial_code/video/optical_flow/optical_flow_dense.py:12]: got it, i change it right now

File: samples/python/tutorial_code/video/optical_flow/optical_flow.py (hunk 1 above)
Comment (UncleLLD) [reply samples/python/tutorial_code/video/optical_flow/optical_flow.py]: got it, i have changed it according to the above code example, please check it, thanks

**Comment by UncleLLD** [comment]
i have changed the above code according to your advices,  thanks

**Review by alalek — APPROVED** [review]
Looks good to me! Thank you :+1:
